check.py

```diff
@@ -125,16 +125,29 @@
           cmd += ['--wasm-only']
         wasm = os.path.join(options.binaryen_test, wasm)
         print '..', asm, wasm
-        actual = run_command(cmd)
 
-        # verify output
-        if not os.path.exists(wasm):
-          fail_with_error('output .wast file %s does not exist' % wasm)
-        expected = open(wasm, 'rb').read()
-        if actual != expected:
-          fail(actual, expected)
-
-        binary_format_check(wasm, verify_final_result=False)
+        def do_asm2wasm_test():
+          actual = run_command(cmd)
+
+          # verify output
+          if not os.path.exists(wasm):
+            fail_with_error('output .wast file %s does not exist' % wasm)
+          expected = open(wasm, 'rb').read()
+          if actual != expected:
+            fail(actual, expected)
+
+          binary_format_check(wasm, verify_final_result=False)
+
+        # test both normally and with pass debug (so each inter-pass state is validated)
+        old_pass_debug = os.environ.get('BINARYEN_PASS_DEBUG')
+        try:
```

Comment:
it's true we don't catch it, but cleaning up the env var change is nice since the exception might be caught higher up